Controllers/OrderController.js

```diff
@@ -1,31 +1,44 @@
 const DbService = require('../Services/DbService')
-const DishesService = require('../Services/DishesService')
-const ObjectId = require('mongodb').ObjectId
+const OrderService = require('../Services/OrderService')
+const JSONResponseService = require('../Services/JSONResponseService')
+
+const orderValidate = require('../Validators/newOrderValidator.json')
+const Ajv = require('ajv')
+const addFormats = require('ajv-formats')
+
+const ajv = new Ajv()
+addFormats(ajv)
 
 let createNewOrder = (req, res) => {
     DbService.connectToDb(async (db) => {
         const order = {
             name: req.body.name,
-            deliveryAddress: req.body.deliveryAddress,
+            firstLineOfAddress: req.body.firstLineOfAddress,
+            postcode: req.body.postcode,
             email: req.body.email,
-            orderItems: req.body.orderItems
         }
-        try {
-            const newOrder = await OrderService.createNewOrder(db, order)
-            if (newOrder.insertedCount === 1) {
-                return res.json({
-                    "success": true,
-                    "message": "Order created",
-                    "status": 200,
-                    "data": newOrder
-                })
+
+        const newOrderValidate = ajv.compile(orderValidate)
+        const valid = newOrderValidate(order)
+        if (valid) {
+            try {
+                const newOrder = await OrderService.createNewOrder(db, order)
+                if (newOrder.insertedCount === 1) {
+                    let response  = JSONResponseService.generateSuccessResponse()
+                    response.message = "Order created"
+                    response.data = newOrder.ops
+                    res.json(response)
+                }
+            } catch (e) {
+                let response = JSONResponseService.generateFailureResponse()
+                response.message = "Database request failed"
+                res.json(response)
```

Comment:
Suggest popping a return statement before your res.jsons in here, otherwise it might do weird stuff with the res.json outside of the if statement